packit_service/worker/database.py

```diff
@@ -23,3 +33,94 @@ def discard_old_srpm_build_logs():
         )
         build.set_logs(None)
         build.set_url(None)
+
+
+def gzip_file(file: Path) -> None:
+    """Gzip compress given file into {file}.gz
+
+    Args:
+        file: File to be compressed.
+    Raises:
+        OSError: If the 'file' can't be opened.
+    """
+    try:
+        with file.open(mode="rb") as f_in:
+            with gzip_open(f"{file}.gz", mode="wb") as f_out:
+                logger.info(f"Compressing {file} into {file}.gz")
+                copyfileobj(f_in, f_out)
+    except OSError as e:
+        logger.error(e)
+        raise
+
+
+def upload_to_s3(
+    file: Path, bucket: str = f"arr-packit-${getenv('DEPLOYMENT', 'dev')}"
+) -> None:
+    """Upload a file to an S3 bucket.
+
+    Args:
+        file: File to upload.
+        bucket: Bucket to upload to.
+    """
+
+    s3_client = boto3_client("s3")
+    try:
+        logger.info(f"Uploading {file} to S3 ({bucket})")
+        s3_client.upload_file(str(file), bucket, file.name)
+    except ClientError as e:
+        logger.error(e)
+        raise
+
+
+def is_aws_configured() -> bool:
+    # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#environment-variables
+    return bool(getenv("AWS_ACCESS_KEY_ID") and getenv("AWS_SECRET_ACCESS_KEY"))
+
+
+def dump_to(file: Path):
+    """Dump 'packit' database into a file.
+
+    To restore db from this file, run:
+    psql -d packit < database_packit.sql
+
+    Args:
+        file: File where to put the dump.
+    Raises:
+        PackitCommandFailedError: When pg_dump fails.
+    """
+    # We have to specify libpq connection string to be able to pass the
+    # password to the pg_dump. Luckily get_pg_url() does almost what we need.
+    pg_connection = get_pg_url().replace("+psycopg2", "")
+    cmd = ["pg_dump", f"--file={file}", f"--dbname={pg_connection}"]
+
+    packit_logger = getLogger("packit")
+    was_debug = False
+    if packit_logger.level == DEBUG:
+        was_debug = True
+        # Temporarily increase log level to avoid password leaking into logs
+        packit_logger.setLevel(INFO)
```

Comment:
Can we use `try-finally` instead?

Reply:
try what, the `setLevel()` or  `run_command()` or both? I don't see how it would work. Can you show me your suggestion as code? thanks

Reply:
Sorry for the lack of explanation -- I was worried that if `run_command` fails, we don't switch the logger back. So, I suggested using `finally` block to be sure it's executed. So, something like this:

```python
was_debug = packit_logger.level == DEBUG
try:
    if was_debug:
        packit_logger.setLevel(INFO)

    db_name = getenv("POSTGRESQL_DATABASE")
    logger.info(f"Running pg_dump to create '{db_name}' database backup")
    run_command(cmd=cmd)

finally:
    if was_debug:
        packit_logger.setLevel(DEBUG)
```

Reply:
I see, thanks, amended.